### ISSUE TITLE
YOLOv3中训练自己的数据集时可以调用官方的yolov3.weights,在它的基础上进行训练吗？

### ISSUE THREAD

**Issue by monoloxo** [YunYang1994/TensorFlow2.0-Examples#87]
如题，训练自己的数据集时初始loss很大，请教如何在yolov3.weights的基础上进行训练？

**Comment by wind754203900** [comment]
用model.load_weights函数加载权重再训练呗

**Comment by forever208** [comment]
同问，训练自己的模型的时候，backbone 部分的权重为啥不直接加载官方已经训练好的权重呢

**Comment by coyote000** [comment]
可以的，但需要在加载模型的时候注释掉原有的模型加载方式，使用如45行的方式加载

------------------&nbsp;原始邮件&nbsp;------------------
发件人:&nbsp;"forever208"<notifications@github.com&gt;;
发送时间:&nbsp;2020年5月30日(星期六) 晚上10:29
收件人:&nbsp;"YunYang1994/TensorFlow2.0-Examples"<TensorFlow2.0-Examples@noreply.github.com&gt;;
抄送:&nbsp;"Subscribed"<subscribed@noreply.github.com&gt;;
主题:&nbsp;Re: [YunYang1994/TensorFlow2.0-Examples] YOLOv3中训练自己的数据集时可以调用官方的yolov3.weights,在它的基础上进行训练吗？ (#87)





 
同问，训练自己的模型的时候，backbone 部分的权重为啥不直接加载官方已经训练好的权重呢

**Comment by forever208** [comment]
> 可以的，但需要在加载模型的时候注释掉原有的模型加载方式，使用如45行的方式加载
> […](#)
> ------------------&nbsp;原始邮件&nbsp;------------------ 发件人:&nbsp;"forever208"<notifications@github.com&gt;; 发送时间:&nbsp;2020年5月30日(星期六) 晚上10:29 收件人:&nbsp;"YunYang1994/TensorFlow2.0-Examples"<TensorFlow2.0-Examples@noreply.github.com&gt;; 抄送:&nbsp;"Subscribed"<subscribed@noreply.github.com&gt;; 主题:&nbsp;Re: [YunYang1994/TensorFlow2.0-Examples] YOLOv3中训练自己的数据集时可以调用官方的yolov3.weights,在它的基础上进行训练吗？ (#87) 同问，训练自己的模型的时候，backbone 部分的权重为啥不直接加载官方已经训练好的权重呢 — You are receiving this because you are subscribed to this thread. Reply to this email directly, view it on GitHub, or unsubscribe.

Hello，45行的方式具体指的是什么？ 方便帮忙解答一下嘛？感谢！

**Comment by llllala** [comment]
可以在train.py  'model = tf.keras.Model(input_tensor, output_tensors)'这句话后面（在40行左右）加上
model.load_weights("./yolov3_myself")

**Comment by forever208** [comment]
感谢 llllala，另外想再问一下，加载权重后准备训练，我想在训练阶段 freeze bn 层，请问需要咋操作呢。bn 的子类我看得不太明白，还请赐教。感恩~！

**Comment by Mr-yun** [comment]
您好,您添加model.load_weights("./yolov3_myself")是否正常运行,我这边添加完后会提示以下错误
./yolov3.weights: Data loss: not an sstable (bad magic number): perhaps your file is in a different file format and you need to use a different restore operator?

**Comment by Oceania2018** [comment]
@Mr-yun @monoloxo 
应该是util工具类来加载:
```python
model = tf.keras.Model(input_tensor, output_tensors)
utils.load_weights(model, "./yolov3.weights")
```

**Comment by wangyaxin1998** [comment]
> 可以在train.py 'model = tf.keras.Model(input_tensor, output_tensors)'这句话后面（在40行左右）加上
> model.load_weights("./yolov3_myself")

您好，model.load_weights("./yolov3.weights")，显示Unable to open file (file signature not found)改怎么解决

**Comment by mastergao57** [comment]
> 可以在train.py 'model = tf.keras.Model(input_tensor, output_tensors)'这句话后面（在40行左右）加上 model.load_weights("./yolov3_myself")

这样就可以了吗？还需要修改别的地方吗？
